airflow/utils/db_manager.py

```diff
@@ -0,0 +1,176 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.module_loading import import_string
+
+if TYPE_CHECKING:
+    from sqlalchemy import MetaData
+
+
+class AttributeCheckerMeta(type):
+    """Metaclass to check attributes of subclasses."""
+
+    def __new__(cls, name, bases, dct):
+        """Check that subclasses are setting the required attributes."""
+        required_attrs = ["metadata", "migration_dir", "alembic_file", "version_table_name"]
+        for attr in required_attrs:
+            if attr not in dct:
+                raise AttributeError(f"{name} is missing required attribute: {attr}")
+        return super().__new__(cls, name, bases, dct)
+
+
+class BaseDBManager(LoggingMixin, metaclass=AttributeCheckerMeta):
```

Comment:
Can this be an AbstractBaseClass instead of rolling our own AttributeCheckMeta?

Reply:
Done